### ISSUE TITLE
Аннтоация диагностики. Расширение

### ISSUE THREAD

**Issue by theshadowco** [1c-syntax/bsl-language-server#362]
Возможно стоит для диагностики добавить аннотации
 - перечень ссылок на источники
 - код правила АПК

**Comment by pimchest** [comment]
Насколько я понимаю это всё лежит в АПК, описание правила, ссылка и т.д. И это необходимо доработать обработку по выгрузке проверок - https://github.com/otymko/acc-export

**Comment by nixel2007** [comment]
@pimchest, не совсем. Для встроенных в bsl ls правил описание и ссылки лежат в маркдауне, служебная мета-информация - в аннотации к джава классу

**Comment by pimchest** [comment]
@nixel2007 то есть имеется ввиду, к встроенным проверкам ls добавить связь с правилами из АПК. Дополнительно добавить ссылку на источник правила, например с ИТС, где их еще нету, так ?

**Comment by nixel2007** [comment]
Да, но не в маркдаун доке, а прям как часть метаданных самой диагностики.

**Comment by nixel2007** [comment]
Я пока не сформировал мнения на этот счёт. Discuss, что называется

**Comment by artbear** [comment]
@theshadowco 1 Ссылки на источники можно выделить в аннотации. 
потом можно будет сделать контроль этих ссылок, анализ и т.п.

2 аннотации по правилам АПК 
- либо вообще не делать
- либо сразу сделать возможность учета нескольких видов подобных ссылок
 - АПК
 - ЕДТ
 - что-то другое - вдруг появится )

**Comment by artbear** [comment]
@nixel2007 @theshadowco я бы еще добавил аннотацию по типу приложения - тонкий, обычное приложение, толстый УФ, веб
вроде бы такой нет

**Comment by nixel2007** [comment]
Обсуждали вчера. Была идея добавить номер стандарта.

**Comment by artbear** [comment]
@nixel2007 к сожалению, только на номер стандарта опираться не получится.
есть разные ссылки на сайте 1С, есть полезные ссылки и с других сайтов.
зачем ограничиваться только стандартами?

**Comment by nixel2007** [comment]
Я не вижу практического смысла в добавлении ссылок на источники в метадату. Чем это может быть нужно, кроме как в мд?

**Comment by artbear** [comment]
- валидация ссылок на доступность
- валидация ссылок на изменение
  - изменилось описание стандарта, стоит взглянуть на реализацию, дополнить\исправить?
- строить перекрестные ссылки между правилами и показывать их
   - я в Пуле с ребятами начинал это делать, но только руками )
 - виджет в СонарКуб
 - виджет в ВСЦ\ЕДТ )
